python/lsst/ts/ATHexapod/controller.py

```diff
@@ -117,28 +141,20 @@ def is_connected(self):
             self.writer.write(f"{cmd}\n".encode())
             await self.writer.drain()
 
-            response = b""
-
             if has_response:
                 try:
-                    keep_going = multi_line
-                    while True:
-                        c = await asyncio.wait_for(self.reader.read(1), timeout=2.)
-                        response += c
-
-                        if c == b'\n' and not keep_going:
-                            break
-                        elif c == b' ':
-                            keep_going = True
-                        elif c == b'\n':
-                            keep_going = False
+                    replies = []
+                    for i in range(num_line):
+                        raw_line = await self.reader.readuntil(b'\n')
+                        line = raw_line.decode("ISO-8859-1").strip()
+                        self.log.debug(f"Read {i+1} of {num_line} lines: {line}")
+                        replies.append(line)
                 except asyncio.TimeoutError:
                     self.log.warning("Timed out waiting for response from controller. Result "
                                      "may be incomplete.")
+                    raise
```

Comment:
Thank you for adding the raise!

Please consider changing the log message to a warning and removing "Result may be incomplete", because now no replies will be returned (there is no way to access the incomplete reply).

How did that work out in practice? Did the raise cause any problems for calling code?